lib/ansible/modules/cloud/amazon/s3_bucket.py

```diff
@@ -155,6 +156,95 @@ def create_tags_container(tags):
     return tags_obj
 
 
+def compare_policy_statement_part(current, new):
+    ''' Takes a matching part of a current policy statement and the
+        new policy statement and compares them. Ensures lists are
+        compared to other lists and calls itself recursively for dicts
+        so nesting structures are as expected.
+        Returns True if the policy has changed.
+    '''
+    if current != new:
+        if isinstance(current, dict):
+            if set(current.keys()) != set(new.keys()):
+                return True
+            for each in set(current.keys()):
+                current_inner = current[each]
+                new_inner = new[each]
+                # recursively checking each value in the dicts
+                if compare_policy_statement_part(current_inner, new_inner):
+                    return True
+        if isinstance(current, list) and isinstance(new, string_types):
+            current = [to_text(each) for each in current]
+            new = [to_text(new)]
+        elif isinstance(new, list) and isinstance(current, string_types):
+            new = [to_text(each) for each in new]
+            current = [to_text(current)]
+        if not isinstance(current, dict) and current != new:
+            return True
+
+    return False
+
+
+def sort_policy_statement(current_policy, new_policy):
+    ''' Takes two policies and looks for matching statement parts (since it is a list of dicts
+        and cannot be easily ordered).
+        Returns a list of tuple pairs that can then be given to
+        compare_policy_statement_part()
+        Return sample:
+          [({
+             "Sid":"AddCannedAcl",
```

Comment:
This example isn't a valid IAM policy, because statement IDs must be unique per-policy http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html#Sid